feature/sessions/src/main/java/io/github/droidkaigi/confsched2023/sessions/component/TimetableGridItem.kt

```diff
@@ -124,6 +149,50 @@ fun TimetableGridItem(
     }
 }
 
+/**
+ *
+ * Calculate the font size of the title by the height of the displayed session grid.
+ *
+ * @param textStyle session title text style.
+ * @param localDensity local density.
+ * @param gridItemHeightPx session grid item height. (unit is px.)
+ * @param speaker session speaker.
+ * @param titleLength session title length.
+ *
+ */
+private fun calculateTitleFontSize(
+    textStyle: TextStyle,
+    localDensity: Density,
+    gridItemHeightPx: Int,
+    speaker: TimetableSpeaker?,
+    titleLength: Int,
+): TextUnit {
+    val titleToScheduleSpaceHeight = with(localDensity) { 4.dp.toPx() }
+    val scheduleHeight = with(localDensity) { 16.dp.toPx() }
+    val scheduleToSpeakerSpaceHeight = with(localDensity) { 16.dp.toPx() }
+    val horizontalPadding = with(localDensity) { (12 * 2).dp.toPx() }
+
+    // The height of the title that should be displayed.
+    var displayTitleHeight =
+        gridItemHeightPx - titleToScheduleSpaceHeight - scheduleHeight - scheduleToSpeakerSpaceHeight - horizontalPadding
+    displayTitleHeight -= if (speaker != null) with(localDensity) { 32.dp.toPx() } else 0f
+
+    // Actual height of displayed title.
+    val boxWidthWithoutPadding = with(localDensity) { (192 - 12 * 2).dp.toPx() }
+    val fontSizePx = with(localDensity) { textStyle.fontSize.toPx() }
+    val textLengthInRow = (boxWidthWithoutPadding / fontSizePx).roundToInt()
+    val rows = titleLength / textLengthInRow
+    val actualTitleHeight = rows * fontSizePx
```

Comment:
[Ask For Option]

I set up the logic as follows.

1. I checked the height of the Box that displays the session title. (Lines 170-178 of the code)
2. Based on the actual length of the session title, default font size, and width of the Box, I check whether it fits within the height I checked earlier.
3. I set the acceptable font size if it fits, or set the minimum font size if it doesn't fit.

It seems to be a little easier to calculate, but I couldn't come up with an opinion myself.

If you don't mind, I would appreciate it if you could give me your opinion.🙇‍♂️